src/http/router.rs

```diff
@@ -154,9 +154,38 @@ pub type Params = Captures<'static, 'static>;
 /// # fn handle_single_segment(req: Request, params: Params) -> anyhow::Result<Response> { todo!() }
 /// # fn handle_exact(req: Request, params: Params) -> anyhow::Result<Response> { todo!() }
 /// ```
+///
+/// Route based on the trailing segment of a Spin wildcard route, instead of on the full path
```

Comment:
I know this follows an existing pattern, but it somehow feels weird that the example code here is on the struct itself instead of on the method. But maybe I would feel the same way if it were the other way around?

Reply:
@rylev Good point.  My rough thinking is that these don't document `handle()` or `suffix()` in isolation, but rather the _interaction_ between `new()/suffix()`, `get/post/put/etc`, and `handle()` - that is, they are examples of "how to use the Router struct", rather than an individual method in isolation.  It does lead to a rather large and hard to navigate docs block though.

And there's something to be said for _also_ having examples on individual methods, because a reader might have an anchored link that takes them directly to a method and bypasses the struct docs.

For now, I'm going to punt on this for this PR, and we can re-assess and/or try different things as part of a separate examples pass.